pkg/controller/data_upload_controller.go

```diff
@@ -828,6 +830,16 @@ func (r *DataUploadReconciler) setupExposeParam(du *velerov2alpha1api.DataUpload
 			return nil, errors.Wrapf(err, "failed to get PVC %s/%s", du.Spec.SourceNamespace, du.Spec.SourcePVC)
 		}
 
+		// check if there is a mapping for source pvc storage class in backupPVC config
+		// if the mapping exists then use the values(storage class, readOnly accessMode)
+		// for backupPVC (intermediate PVC in snapshot data movement) object creation
+		backupPVCStorageClass := du.Spec.CSISnapshot.StorageClass
+		readOnly := false
+		if value, exists := r.backupPVCConfig[du.Spec.CSISnapshot.StorageClass]; exists {
+			backupPVCStorageClass = value.StorageClass
+			readOnly = value.ReadOnly
```

Comment:
`readOnly = value.ReadOnly` should be outside of the if branch which checks `r.backupPVCConfig[du.Spec.CSISnapshot.StorageClass]`